src/components/list_group/pinnable_list_group/pinnable_list_group.tsx

```diff
@@ -115,7 +115,7 @@ export const OuiPinnableListGroup: FunctionComponent<OuiPinnableListGroupProps>
       );
 
       // Add the pinning action unless the item has it's own extra action
-      if (onPinClick && !itemProps.extraAction && pinnable) {
```

Comment:
Why is `onPinClick` removed here? From a brief look at the code, it seems like it should be null checked

Reply:
The `onPinClick` is defined as required prop for `OuiPinnableListGroup` component. Typescript will give error, if `onPinClick` is not provided. So no extra validation needed. I hope we can benefit from type checking here

Reply:
Compile-time warning cannot take the place of proper guarding. There are many scenarios that TS will not be able check during compiling.

That said, `onPinClick` check shouldn't be up there but should be added down.
```
if (onPinClick) itemProps.extraAction.onClick = () => onPinClick(item);
```

Reply:
I committed the above suggestion.